### PR TITLE
Post-merge review addressed

### PR DESCRIPTION
**Pull request by ymesika**
Addressing reviewing comments made by @hzxuzhonghu after #10036 got merged.

### CODE REVIEW

**Comment by ymesika** [comment]
@hzxuzhonghu approved?

**Comment by hzxuzhonghu** [comment]
I have no permission for /pkg

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -113,8 +113,7 @@ var (
 		plugin.Snidnat,
 	}
 
-	// Global mutex
-	globalMutex = sync.Mutex{}
+	runOnce = sync.Once{}
```
Comment (ozevren) [inline pilot/pkg/bootstrap/server.go]: It is not clear what this runOnce does. A better name could be enableHistogramOnce

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -1136,12 +1133,10 @@ func (s *Server) grpcServerOptions() []grpc.ServerOption {
...
 	}
 
-	// EnableHandlingTimeHistogram() has a data race within it (reading/writing to its
-	// serverHandledHistogramEnabled). Using a package mutex to avoid data races.
-	// Should be re-examined with an updated revision of the go-grpc-prometheus.
-	globalMutex.Lock()
-	prometheus.EnableHandlingTimeHistogram()
-	globalMutex.Unlock()
+	// Running the prometheus global func once to avoid data races within it
+	runOnce.Do(func() {
```
Comment (ozevren) [inline pilot/pkg/bootstrap/server.go]: The placement of the method call itself seems wrong. We're enabling a global setting within what looks like a pure function that generates option values. 

Perhaps placing it at a higher call point will help with the race condition itself as well?

Comment (ymesika) [reply pilot/pkg/bootstrap/server.go]: Good idea

**Pull request merged**